orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -110,15 +112,15 @@ def __init__(self, module, debug_options: DebugOptions, fallback_manager: _Fallb
                 lambda x, y: x | y,
                 [_SkipCheck[name] for name in _utils.parse_os_env_skip_check_flags("ORTMODULE_SKIPCHECK_POLICY")],
             )
-        self._first_skip_check_warning = True
+        self._first_skip_check_warning: bool = True
```

Comment:
Shall we only annotate at function level? In case the type changes, it might become hard to maintain if the type hints are everywhere.

Reply:
Hmmm, maybe you are right, we should not bring too much burden to maintain every line of code. Let's me remove those. :)

Reply:
I have remove almost all local type hints especially those primitive types easy for us to understand. For some important data structures, still give a type hints for them, this will help us to understand the variable intention also. Hope it makes sense. BTW, just force pushed the commits.